lib/media/segment_prefetch.js

```diff
@@ -433,6 +442,9 @@ shaka.media.SegmentPrefetchOperation = class {
             buffered = new Uint8Array(0);
           }
         });
+    return new Promise((resolve, reject) => {
+      this.operation_.chain(resolve, reject);
```

Comment:
I think instead of wrapping with .chain, you can just return this.operation_.promise, right?

Reply:
@theodab can you review it? Thanks!

Reply:
You're right, that's better.